### PR TITLE
Gitbit weight query can now get more than 31 dyas of data

### PR DESCRIPTION
**Pull request by ntrous**

### CODE REVIEW

File: shim-server/src/main/java/org/openmhealth/shim/fitbit/FitbitShim.java [hunk 1]
```diff
@@ -196,10 +196,39 @@ protected String getAuthorizationUrl(UserRedirectRequiredException exception) {
         OffsetDateTime currentDate = startDate;
 
         if (usesDateRangeQuery(fitbitDataType)) {
+    		if(fitbitDataType != fitbitDataType.WEIGHT) {
```
Comment (JCWardle) [inline shim-server/src/main/java/org/openmhealth/shim/fitbit/FitbitShim.java:199]: So the date limit only applys to weight?

File: shim-server/src/main/java/org/openmhealth/shim/fitbit/FitbitShim.java [hunk 2]
```diff
@@ -196,10 +196,39 @@ protected String getAuthorizationUrl(UserRedirectRequiredException exception) {
...
+                 */
+    			OffsetDateTime tempEndDate = endDate;
+            	long daysBetween = java.time.temporal.ChronoUnit.DAYS.between(currentDate, endDate);
+            	if(daysBetween > 31){
+            		tempEndDate = currentDate.plusDays(30);
+            	}
+            	
+    			List<ShimDataResponse> responses = new ArrayList<>();
+
+                while (currentDate.toLocalDate().isBefore(endDate.toLocalDate()) ||
```
Comment (JCWardle) [inline shim-server/src/main/java/org/openmhealth/shim/fitbit/FitbitShim.java:217]: Seems good for now. We might want to consider threading these requests later?

File: shim-server/src/main/java/org/openmhealth/shim/fitbit/FitbitShim.java (hunk 1 above)
Comment (ntrous) [reply shim-server/src/main/java/org/openmhealth/shim/fitbit/FitbitShim.java:199]: Yes that's correct!

File: shim-server/src/main/java/org/openmhealth/shim/fitbit/FitbitShim.java (hunk 2 above)
Comment (ntrous) [reply shim-server/src/main/java/org/openmhealth/shim/fitbit/FitbitShim.java:217]: Indeed, we can sort that in the rewrite.

**Pull request merged**